cmake/JuliaDeps.cmake

```diff
@@ -0,0 +1,16 @@
+jl_str_option(LLVM_VER "" 3.3)
+jl_str_option(LLVM_LIB_SUFFIX "" "")
+jl_str_option(PCRE_VER "" 10.10)
+jl_str_option(DSFMT_VER "" 2.2.3)
+jl_str_option(LAPACK_VER "" 3.5.0)
+jl_str_option(ARPACK_VER "" 3.2.0)
+jl_str_option(FFTW_VER "" 3.3.4)
+jl_str_option(SUITESPARSE_VER "" 4.4.2)
+jl_str_option(UNWIND_VER "" 1.1)
+jl_str_option(OSXUNWIND_VER "" 0.0.1)
+jl_str_option(GMP_VER "" 6.0.0)
+jl_str_option(MPFR_VER "" 3.1.2)
+jl_str_option(PATCHELF_VER "" 0.8)
+jl_str_option(GIT_VER "" 1.8.5.6)
+jl_str_option(VIRTUALENV_VER "" 1.11.6)
+jl_str_option(RMATH_JULIA_VER "" 0.1)
```

Comment:
Probably too hacky, but I had thrown together this at one point to make it possible for cmake and the existing `deps/Versions.make` and `Make.user` to coexist for a little while without having to duplicate this info:
```
if(CMAKE_VERSION VERSION_GREATER 3)
  set_source_files_properties(Make.user deps/Versions.make
    PROPERTIES CMAKE_CONFIGURE_DEPENDS)
endif()

macro(read_settings_file FILENAME)
  get_filename_component(ABSPATH ${FILENAME} ABSOLUTE)
  if(EXISTS ${ABSPATH})
    file(STRINGS ${ABSPATH} CONTENTS NEWLINE_CONSUME)
    string(REPLACE " =" "=" CONTENTS ${CONTENTS})
    string(REPLACE "= " "=" CONTENTS ${CONTENTS})
    string(REPLACE "\n" ";" CONTENTS ${CONTENTS})
    foreach(LINE ${CONTENTS})
      string(REGEX MATCH "^#.*" ISCOMMENT ${LINE})
      if(NOT ISCOMMENT)
        string(REGEX REPLACE "^override " "" LINE ${LINE})
        string(REPLACE "=" ";" LINE ${LINE})
        list(LENGTH LINE LINELENGTH)
        if(LINELENGTH LESS 3)
          list(GET LINE 0 VAR)
          list(GET LINE 1 VALUE)
          if(DEFINED ${VAR})
            message(STATUS "${FILENAME}: ignored setting of ${VAR} = ${VALUE}")
          else()
            set(${VAR} ${VALUE} CACHE STRING "${VAR}")
            message(STATUS "${VAR} = ${VALUE}")
          endif()
        else()
          message(WARNING "${FILENAME}: line ${LINE} had too many elements")
        endif()
      endif()
    endforeach(LINE)
  endif()
endmacro()

# priority: first from command-line -D settings, then from Make.user, then from deps/Versions.make
read_settings_file(Make.user)
read_settings_file(deps/Versions.make)
```